pkg/odo/genericclioptions/util.go

```diff
@@ -34,32 +31,40 @@ func getFirstChildOfCommand(command *cobra.Command) *cobra.Command {
 
 // checkProjectCreateOrDeleteOnlyOnInvalidNamespace errors out if user is trying to create or delete something other than project
 // errFormatForCommand must contain one %s
-func checkProjectCreateOrDeleteOnlyOnInvalidNamespace(command *cobra.Command, errFormatForCommand string) {
+func checkProjectCreateOrDeleteOnlyOnInvalidNamespace(command *cobra.Command, errFormatForCommand string) error {
 	// do not error out when its odo delete -a, so that we let users delete the local config on missing namespace
 	if command.HasParent() && command.Parent().Name() != "project" && (command.Name() == "create" || (command.Name() == "delete" && !command.Flags().Changed("all"))) {
 		err := fmt.Errorf(errFormatForCommand, command.Root().Name())
-		util.LogErrorAndExit(err, "")
+		if err != nil {
+			return fmt.Errorf(errFormatForCommand, command.Root().Name())
+		}
 	}
+	return nil
 }
```

Comment:
```suggestion
func checkProjectCreateOrDeleteOnlyOnInvalidNamespace(command *cobra.Command, errFormatForCommand string) error {
	// do not error out when its odo delete -a, so that we let users delete the local config on missing namespace
	if command.HasParent() && command.Parent().Name() != "project" && (command.Name() == "create" || (command.Name() == "delete" && !command.Flags().Changed("all"))) {
		return fmt.Errorf(errFormatForCommand, command.Root().Name())
	}
	return nil
}
```